src/tooltip/src/Tooltip.js

```diff
@@ -80,14 +87,28 @@ export default class Tooltip extends PureComponent {
   show = () => {
     if (this.state.isShown) return
     this.setState({
-      isShown: true
+      willShow: true
     })
+    setTimeout(() => {
+      if (!this.state.willShow) return
+      this.setState({
+        isShown: true
+      })
+    }, this.props.showDelay)
   }
 
   hide = () => {
-    if (!this.state.isShown) return
+    if (!this.state.isShown) {
+      if (!this.state.willShow) return
+      this.setState({
+        willShow: false
+      })
+      return
+    }
```

Comment:
I think we could probably simplify this? Looks like every result of this function ends with `isShown: false, willShow: false` right?

```js
// nothing to do
if (!this.state.isShown && !this.state.willShow) {
  return
}

// Otherwise, set both the current and eventual shown states to false
this.setState({
  isShown: false,
  willShow: false
})
```

You could also do this with the same effect:
```js
hide = () => {
  this.setState({
    isShown: false,
    willShow: false
  })
}
```

It'll end up calling setState in more cases, but the resulting state will not change from what the current code is doing.

Reply:
That's a great point! I added a commit to simplify it 👍 

I went with the second solution of just setting state every time `hide` is called. It's not likely for `hide` to be called while `!this.state.isShown && !this.state.willShow`, since the mouse will always enter before leaving. (unless the user rapidly flails their mouse over the tooltip, but even that is fine because we're using PureComponent)